LoopKitUI/SupportUI.swift

```diff
@@ -27,26 +27,50 @@ public struct LoopScenario: Hashable {
     }
 }
 
-public protocol SupportUIDelegate: AlertIssuer { }
+public enum SettingsMenuSection: Equatable {
+    case configuration
+    case support
+    case custom(localizedTitle: String)
+
+    public var customLocalizedTitle: String? {
+        switch self {
+        case .custom(let title):
+            return title
+        default:
+            return nil
+        }
+    }
+}
+
+public struct CustomMenuItem {
+    public let section: SettingsMenuSection
+    public let view: AnyView
```

Comment:
`AnyView` erases all identity from a SwiftUI View and can cause needless re-renders. Could this struct instead be generic over the `View` type?

```
public struct CustomMenuItem<V: View> {
    public let section: SettingsMenuSection
    public let view: V
```

Reply:
Yes, I would love to avoid this and have tried a number of times. But because this is vended from a protocol type, it has proved very challenging, and I actually haven't found a way around it. Would welcome you to take a swing at it. This PR does not introduce the situation of returning an `AnyView`; it just returns it wrapped in a type.

Reply:
I did try that and then encountered problems where it's being used:

<img width="777" alt="Screenshot 2023-05-20 at 9 33 50 AM" src="https://github.com/tidepool-org/LoopKit/assets/14649/d101d872-b0ae-4eea-8b98-5e2ab5d44059">

Not exactly sure what the solution to this is either. I tried declaring `view: any (View & AccessibilityRotorContent)`, but  that was breaking the compiler, lol.

I think we should set aside some time to look into addressing this: https://tidepool.atlassian.net/browse/LOOP-4652